src/vellum/workflows/nodes/experimental/tool_calling_node/utils.py

```diff
@@ -0,0 +1,89 @@
+from typing import Iterator, List, Optional, Type
+
+from vellum import ChatMessage, FunctionDefinition, PromptBlock
+from vellum.workflows.nodes.bases import BaseNode
+from vellum.workflows.nodes.displayable.inline_prompt_node.node import InlinePromptNode
+from vellum.workflows.outputs.base import BaseOutput
+from vellum.workflows.ports.port import Port
+from vellum.workflows.references.lazy import LazyReference
+from vellum.workflows.types.core import EntityInputsInterface
+
+
+class FunctionNode(BaseNode):
+    """Node that executes a specific function."""
+
+    function: FunctionDefinition
+
+    def run(self) -> BaseNode.Outputs:
+        # TODO: We should think about how to execute the function
```

Comment:
[oos] this should already be ticketed in you project, surrounding the invocations of custom inline functions.